docs/REST API/rest-api.md

```diff
@@ -6182,6 +6187,195 @@ Content-Length: <length>
+
+`GET /api/v1/fleet/setup_experience/software`
+
+| Name  | Type   | In    | Description                              |
+| ----- | ------ | ----- | ---------------------------------------- |
+| team_id | integer | query | _Available in Fleet Premium_. The ID of the team to filter software by. If not specified, it will filter only software that's available to hosts with no team. |
+| page | integer | query | Page number of the results to fetch. |
+| per_page | integer | query | Results per page. |
+
+
+#### Example
+
+`GET /api/v1/fleet/setup_experience/software?team_id=3`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "software_titles": [
+    {
+      "id": 1,
+      "name": "Google Chrome.app",
+      "versions_count": 3,
+      "source": "apps",
+      "browser": "",
+      "hosts_count": 10,
+      "installed_during_setup": true,
+      "versions": [
+        {
+          "id": 101,
+          "version": "1.0.0",
+          "vulnerabilities": ["CVE-2021-12345", "CVE-2021-67890"]
+        },
+        {
+          "id": 102,
+          "version": "1.1.0",
+          "vulnerabilities": []
+        }
+      ],
+      "software_package": {
+        "name": "Software A Package",
+        "version": "1.0.0",
+        "self_service": true
+      },
+      "app_store_app": {
+        "app_store_id": 201,
+        "version": "1.0.0"
+      }
+    }
+  ],
+  {
+    "count": 2,
+    "counts_updated_at": "2024-10-04T10:00:00Z",
+    "meta": {
+      "has_next_results": false,
+      "has_previous_results": false
+    }
+  },
+}
+```
+
+### Update setup experience software
+
+_Available in Fleet Premium_
+
+Set software that will be automatically installed during macOS setup. Software that isn't included in the request will be unset.
+
+`PUT /api/v1/fleet/setup_experience/software`
+
+| Name  | Type   | In    | Description                              |
+| ----- | ------ | ----- | ---------------------------------------- |
+| team_id | integer | query | _Available in Fleet Premium_. The ID of the team to set the software for. If not specified, it will set the software for hosts with no team. |
+| software_title_ids | array | body | The ID of software titles to install during macOS setup. |
+
+#### Example
+
+`PUT /api/v1/fleet/setup_experience/software?team_id=3`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "software_title_ids": [23,3411,5032]
+}
+```
+
+### Add setup experience script
+
+_Available in Fleet Premium_
+
+Add a script that will automatically run during macOS setup.
+
+`POST /api/v1/fleet/setup_experience/script`
+
+| Name  | Type   | In    | Description                              |
+| ----- | ------ | ----- | ---------------------------------------- |
+| team_id | integer | query | _Available in Fleet Premium_. The ID of the team to add the script to. If not specified, script will be added for hosts with no team. |
```

Comment:
@marko-lisica just a heads up we should probably make this a form data type like the file. POST methods done tend to use query params and we include this in the form for other requests like uploading a custom profile